nmpolicy/internal/resolver/resolver.go

```diff
@@ -0,0 +1,58 @@
+package resolver
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/nmstate/nmpolicy/nmpolicy/internal/ast"
+)
+
+type Resolver struct {
+	CurrentState map[string]interface{}
+}
+
+func (c *Resolver) Resolve(rootNode ast.Node) (map[string]interface{}, error) {
+	if rootNode.EqFilter != nil {
+		inputSource, err := c.getInputSource(rootNode.EqFilter[0])
+		if err != nil {
+			return nil, resolveError(fmt.Errorf("failed to discover the input source : %v", err))
+		}
+
+		path, err := c.getPath(rootNode.EqFilter[1])
+		if err != nil {
+			return nil, resolveError(fmt.Errorf("failed to discover the path : %v", err))
+		}
+		filteredValue, err := c.filteredValue(rootNode.EqFilter[2])
+		if err != nil {
+			return nil, resolveError(fmt.Errorf("failed to discover the filtered value : %v", err))
+		}
+
+		return filter(inputSource, *path, *filteredValue)
+	}
+	return nil, resolveError(fmt.Errorf("root node has unsupported operation : %v", rootNode))
+}
+
+func (c *Resolver) getInputSource(inputSourceNode ast.Node) (map[string]interface{}, error) {
+	//if inputSourceNode.Identity != nil && *inputSourceNode.Identity == ast.CurrentStateLiteral {
+	if reflect.DeepEqual(inputSourceNode.Terminal, ast.CurrentStateIdentity()) {
+		return c.CurrentState, nil
+	}
```

Comment:
Let's try not to use reflect.

```suggestion
        if inputSourceNode.Identity != nill && *inputSourceNode.Identity == ast.CurrentStateIdentity().Identity
		return c.CurrentState, nil
	}
```

Reply:
@qinqon  Please let me know when you add `func equals` to Terminal and I will use it.